pkg/autodiscovery/providers/kube_endpoints.go

```diff
@@ -144,11 +144,13 @@ func parseServiceAnnotationsForEndpoints(services []*v1.Service) []configInfo {
 			log.Debug("Ignoring a nil service")
 			continue
 		}
-		endptConf, errors := extractTemplatesFromMap(apiserver.EntityForEndpoints(svc.Namespace, svc.Name, ""), svc.Annotations, kubeEndpointAnnotationPrefix)
+		endpointsID := apiserver.EntityForEndpoints(svc.Namespace, svc.Name, "")
```

Comment:
Since we store the value in the `endptConf`, can we test if not already set before doing a call to the apiserver?